storage/merge_test.go

```diff
@@ -1560,7 +1560,8 @@ func TestMergeGenericQuerierWithSecondaries_ErrorHandling(t *testing.T) {
 				}
 			})
 			t.Run("LabelNames", func(t *testing.T) {
-				res, w, err := q.LabelNames(ctx)
+				hints := &LabelHints{Limit: math.MaxInt}
```

Comment:
The only changes we should make to those tests is set the new arg `hints=nil` and everything should keep working as before.
To test the limit hint, I suggest you add new tests/cases.

Reply:
Yes, that makes sense. I would add more tests when I implement the limiting logic inside the `mergeGenericQuerier`. I will create a follow up PR for that soon after this is merged.